internal/cmd/base/servers.go

```diff
@@ -462,16 +462,27 @@ func (b *Server) ConnectToDatabase(dialect string) error {
 	return nil
 }
 
-func (b *Server) CreateDevDatabase(ctx context.Context, dialect string, opt ...Option) error {
-	opts := getOpts(opt...)
+func (b *Server) CreateDevDatabase(ctx context.Context, opt ...Option) error {
 
-	var container, url string
+	var container, url, dialect string
 	var err error
 	var c func() error
 
+	opts := getOpts(opt...)
+
+	// opts.WithContainerImage could have repo:tag, so split it and only
+	// assign the repo to dialect
+	switch opts.withContainerImage {
+	case "postgres":
+		dialect = "postgres"
+	default:
+		separated := strings.Split(opts.withContainerImage, ":")
+		dialect = separated[0]
```

Comment:
I defer to @jimlambrt on this but I think we currently always want the dialect to be "postgres". The dialect is passed through to the SQL driver and is related to the actual SQL that we have. I'm not sure it would actually work if we set the dialect to anything else, even if the docker container itself is launching something that _speaks_ the postgres protocol.

Reply:
I think for now the best way to handle this would be to have a WithDialect option that defaults to `postgres`.

Reply:
I actually don't think we want this anymore but I'm looking